### PR TITLE
[RF] Introducing binned likelihood fit optimization in HistFactory

### PR DESCRIPTION
**Pull request by guitargeek**
In a binned likelihood fit, it is possible to skip the PDF normalization
when the unnormalized binned PDF can be interpreted directly in terms of
event yields. This is now done by default for models HistFactory models,
which results in great speedups for binned fits with many channels. Some
RooFit users like ATLAS were already using this for a long time, so this
feature is battle-tested.

To disable this optimization when using the `hist2workspace` executable,
add the `-disable_binned_fit_optimization` command line argument.
Directly in C++, you can also set the `binnedFitOptimization` to `false`
in the HistFactory configuration as follows:
```C++
RooStats::HistFactory::MakeModelAndMeasurementFast(measurement, {.binnedFitOptimization=false});
```
If your compiler doesn't support aggregate initialization with designators, you
need to create and edit the configuration struct explicitely:
```C++
RooStats::HistFactory::HistoToWorkspaceFactoryFast::Configuration hfCfg;
hfCfg.binnedFitOptimization = false;
RooStats::HistFactory::MakeModelAndMeasurementFast(measurement, hfCfg);
```

### CODE REVIEW

**Review by lmoneta — APPROVED** [review]
LGTM!
Maybe ia nice addition would be a tutorial or a test , it could also be in root bench, showing the difference in time performance in the two cases. It does not need to be a complex model, but just a simple one.

File: README/ReleaseNotes/v628/index.md [hunk 1]
```diff
@@ -90,6 +90,29 @@ Before v6.28, it was ensured that no `RooArgSet` and `RooDataSet` objects on the
...
 Hence, if your code uses pointer comparisons to uniquely identify RooArgSet or RooDataSet instances, please consider using the new `RooArgSet::uniqueId()` or `RooAbsData::uniqueId()`.
 
+### Introducing binned likelihood fit optimization in HistFactory
+
+In a binned likelihood fit, it is possible to skip the PDF normalization when
+the unnormalized binned PDF can be interpreted directly in terms of event
+yields. This is now done by default for models HistFactory models, which
+results in great speedups for binned fits with many channels. Some RooFit users
+like ATLAS were already using this for a long time, so this feature is
+battle-tested.
```
Comment (hageboeck) [inline README/ReleaseNotes/v628/index.md:100]: These kind of claims rot very fast. 🙂

File: README/ReleaseNotes/v628/index.md [hunk 2]
```diff
@@ -90,6 +90,29 @@ Before v6.28, it was ensured that no `RooArgSet` and `RooDataSet` objects on the
 With v6.28, this is not guaranteed anymore.
 Hence, if your code uses pointer comparisons to uniquely identify RooArgSet or RooDataSet instances, please consider using the new `RooArgSet::uniqueId()` or `RooAbsData::uniqueId()`.
 
+### Introducing binned likelihood fit optimization in HistFactory
+
+In a binned likelihood fit, it is possible to skip the PDF normalization when
+the unnormalized binned PDF can be interpreted directly in terms of event
+yields. This is now done by default for models HistFactory models, which
```
Comment (hageboeck) [inline README/ReleaseNotes/v628/index.md:97]:
```suggestion
yields. This is now done by default for HistFactory models, which
```

File: roofit/histfactory/inc/RooStats/HistFactory/MakeModelAndMeasurementsFast.h [hunk 3]
```diff
@@ -18,11 +19,12 @@
 namespace RooStats{
   namespace HistFactory{
 
-    RooWorkspace* MakeModelAndMeasurementFast( RooStats::HistFactory::Measurement& measurement );
-    //RooWorkspace* MakeModelFast( RooStats::HistFactory::Measurement& measurement );
+    RooWorkspace* MakeModelAndMeasurementFast(
```
Comment (hageboeck) [inline roofit/histfactory/inc/RooStats/HistFactory/MakeModelAndMeasurementsFast.h:22]: See? This name also didn't age well. 🙂

Comment (guitargeek) [reply roofit/histfactory/inc/RooStats/HistFactory/MakeModelAndMeasurementsFast.h:22]: Haha that's the worst name possible, and everyone uses it!

**Comment by guitargeek** [comment]
Hi @lmoneta and @hageboeck, thanks for the feedback!

I will open a PR in rootbench with a benchmark based on the hf001_example.C tutorial.

**Pull request merged**